service/matching/matcher.go

```diff
@@ -79,7 +81,8 @@ const (
 
 var (
 	// Sentinel error to redirect while blocked in matcher.
-	errInterrupted = errors.New("interrupted offer")
+	errInterrupted    = errors.New("interrupted offer")
+	errNoRecentPoller = status.Error(codes.DeadlineExceeded, "no poller seen for task queue recently, worker may be down")
```

Comment:
Now I'm wondering if this should be a failed precondition error since this doesn't wait for the whole deadline to expire.

https://grpc.github.io/grpc/core/md_doc_statuscodes.html